[docs] Add `dbt parse` troubleshooting to dbt reference (#22326)
dagster-io/dagster: docs/content/integrations/dbt/reference.mdx

When deploying your Dagster project to production, **we recommend generating the manifest at build time** to avoid the overhead of recompiling your dbt project every time your Dagster code is executed. A `manifest.json` should be precompiled and included in the Python package for your Dagster code.

Creating the manifest at runtime in production is known to cause issues and is not recommended. This is often caused by the following code and should be avoided.

> ```python startafter=start_troubleshooting_dbt_manifest endbefore=end_troubleshooting_dbt_manifest file=/integrations/dbt/dbt.py dedent=4
> import os

Reviewer: To make this explicit?

```suggestion
```python startafter=start_troubleshooting_dbt_manifest endbefore=end_troubleshooting_dbt_manifest file=/integrations/dbt/dbt.py dedent=4
"""❌ This is not recommended."""
import os
```

And this for the second code snippet there should be a 

```python
"""✅ This is recommended!"""
```

Author: Sure, all changes done in [6614302](https://github.com/dagster-io/dagster/pull/22326/commits/661430260719fc837c59b5824898ff6f9abf6985)

Reviewer: Thanks!

from pathlib import Path

from dagster_dbt import DbtCliResource

dbt_project_dir = Path(__file__).joinpath("..", "..", "..").resolve()
dbt = DbtCliResource(project_dir=os.fspath(dbt_project_dir))
dbt_parse_invocation = dbt.cli(["parse"], target_path=Path("target")).wait()

> dbt_manifest_path = dbt_parse_invocation.target_path.joinpath("manifest.json")

Reviewer: Can we make this very similar in structure to the second code snippet?

```python
dbt_project_dir = Path(__file__).joinpath("..", "..", "..").resolve()
dbt = DbtCliResource(project_dir=os.fspath(dbt_project_dir))

# A manifest will always be created at runtime.
dbt_manifest_path = (
    dbt.cli(
        ["--quiet", "parse"],
        target_path=Path("target"),
    )
    .wait()
    .target_path.joinpath("manifest.json")
)
```

```

> In this example, the command `dbt parse` is invoked at run time. To avoid any issues, it is recommended to create the manifest at build time, like in the following example.

Reviewer: ```suggestion
To avoid any issues, we recommend that the manifest is created at run time only during local development. In production, the manifest should be created at build time.
```
